src/Token.php

```diff
@@ -80,6 +80,9 @@ public function getType(): int
         return $this->type;
     }
 
+    /**
+     * @return string|int|float|bool|null
```

Comment:
This is wrong if the constructor does not have a parameter type restricting the type of the value.